src/cryptonote_core/service_node_rules.cpp

```diff
@@ -280,4 +280,18 @@ bool get_portions_from_percent_str(std::string cut_str, uint64_t& portions) {
   return false;
 }
 
+uint64_t get_atomic_amount_from_percent_str(std::string cut_str) {
+  const auto pct = parse_fee_percent(cut_str);
+  if (!pct)
+    throw "could not parse fee percent";
+
+  if(*pct < 0.0 || *pct > 100.0)
+    throw "fee percent out of bounds";
+
+  auto atomic_amount = static_cast<uint16_t>(std::lround(*pct / 100.0 * cryptonote::STAKING_FEE_BASIS));
+  if (*pct == 100.0)
+    atomic_amount = cryptonote::STAKING_FEE_BASIS;
+
+  return atomic_amount;
+}
```

Comment:
`rpc_command_executor.cpp` should get updated to call this function (it basically has duplication of this logic, which I'm thinking is where it came from).